database2/cassandraModel.js

```diff
@@ -0,0 +1,147 @@
+const cass = require('cassandra-driver');
+
+const client = new cass.Client({
+  contactPoints: ['localhost'],
+  localDataCenter: 'datacenter1',
+  keyspace: 'similar_prod'
+});
+
+
+const getRange = (id, category) => {
+  if (id > 12060000 || category === 'laundry') {
+    // 'laundry': 12060001 - 14070000
+    return [12060001, 14070000];
+  } else if (id > 10050000 || category === 'closet') {
+    // 'closet': 10050001 - 12060000,
+    return [10050001, 12060000];
+  } else if (id > 8040000 || category === 'bathroom') {
+    // 'bathroom' 8040001 - 10050000,
+    return [8040001, 10050000];
+  } else if (id > 6030000 || category === 'bedroom') {
+    // 'bedroom' 6030001 - 8040000,
+    return [6030001, 8040000];
+  } else if (id > 4020000 || category === 'dining') {
+    // 'dining'4020001 - 6030000,
+    return [4020001, 6030000];
+  } else if (id > 2010000 || category === 'kitchen') {
+    // 'kitchen': 2010001 - 4020000,
+    return [2010001, 4020000];
+  } else {
+    //'living-room': 1 - 2010000,
+    return [1, 2010000];
+  }
+};
+
+const getCatId = (category) => {
+  if (category === 'laundry') {
+    return 7;
+  } else if (category === 'closet') {
+    return 6;
+  } else if (category === 'bathroom') {
+    return 5;
+  } else if (category === 'bedroom') {
+    return 4;
+  } else if (category === 'dining') {
+    return 3;
+  } else if (category === 'kitchen') {
+    return 2;
+  } else {
+    return 1;
+  }
+};
+
+client.connect();
+module.exports = {
+  getOne : (request, response) => {
+    console.log("get params received in model: ", request.params);
+    const productId = request.params.id;
+    //const productId = 7;
+    const query = `SELECT * FROM products WHERE productid = ?`;
+    client.execute(query, [productId], {prepare: true})
+    .then(result => {
+      const row = result.rows[0];
+      console.log("price result", row.price)
+      const low = parseInt(row.price) - 50;
+      const high = parseInt(row.price) + 50;
+      const values = getRange(productId, null).concat(low, high);
+      const query = `SELECT * FROM products WHERE productid >= ? AND productid <= ? AND price >= ? AND price <= ? limit 8 allow filtering`;
```

Comment:
This is an interesting solution. I had to look at it a few times to understand what's happening. Could you explain to me how this is working? Based on a given productId, ythis will query for up to 8 products, then return those 8 products, right? This is 'getOne', so if it's working the way I'm reading it, I'm a little confused.

Reply:
You got me on these variable names! Will change to getSimilarProducts!

Reply:
The first query gets the price of the product selected, the second query gets 8 products from the same category that are within $50 of selected product's price.